scripts/bake-ide-ami.ps1

```diff
@@ -468,9 +468,33 @@ $jsonObject = @"
                 $ChefRecipe = "VLWebServer::IDEBaseAzure"
             }
 
+            # 163204: Generate PFX using secret from Azure Vault
             # Make sure the session is initialised correctly
             ReConnect-Session
 
+            # 163204: Generate PFX using secret from Azure Vault
+            if ( $Cloud -eq 'Azure' ) {
+                # ensures that the PFX Certificates are present in the license key path
+                $vmPfxCertificates = @{IntegratorLicensePrivateKey = "LANSAIntegratorLicense"; ScalableLicensePrivateKey = "LANSAScalableLicense"};
+                foreach ($certificateName in $vmPfxCertificates.Keys) {
+                    $secret = Get-AzKeyVaultSecret -VaultName $KeyVault -Name $certificateName
+                    if ( $secret ) {
+                        # Write to a file
+                        Write-Verbose "$(Log-Date) Create a PFX File for $certificateName Certificate"
+    
+                        # contains the JSON object with data, password and data type
+                        $certData = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secret.SecretValueText))| ConvertFrom-Json
```

Comment:
As per the document referred to in Issue #2 , you should be using `Add-AzVmSecret`, not saving the file on your local PC. Your solution is highly insecure and is not following the guidelines.
_The DP is required to enhance the current baking script bake-ide-ami.ps1 and add in Powershell commands to pull the certificates into the VM using `Add-AzVmSecret`. The secret url must be derived from Azure Key Vault and not specified explicitly so that the LATEST secret is obtained. Details are in the tables below. Note that the existing code already does something similar with `bakingWinRMCertificate`._ 

This will also require locating the code that adds the pfx files into the certificate store and deleting it. Note you will need to have your code in the robe070/cookbooks repo. This is why I have made you contributors to that repo and not requested that you fork it.

Reply:
hi @robe070 
Thank you for your inputs. I will push my changes to the robe070/cookbooks so that the VM can pull the updated script. I had removed the pfx files from my local machine securely. I am closing the PR for now.